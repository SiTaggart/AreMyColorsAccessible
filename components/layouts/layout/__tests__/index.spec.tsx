/* eslint-env jest */
/// <reference types="jest" />

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Layout } from '..';

describe('Layout', (): void => {
  it('renders without crashing', (): void => {
    ReactDOM.render(
      <Layout>
        <div>children</div>
      </Layout>,
      document.createElement('div')
    );
    const layoutCmp = renderer
      .create(
        <Layout>
          <div>children</div>
        </Layout>
      )
      .toJSON();
    expect(layoutCmp).toMatchSnapshot();
  });

  it('should set full class with full variant', (): void => {
    const layoutCmp = renderer
      .create(
        <Layout variant="full">
          <div>children</div>
        </Layout>
      )
      .toJSON();
    expect(layoutCmp).toMatchSnapshot();
  });

  it('should set large class with large variant', (): void => {
    const layoutCmp = renderer
      .create(
        <Layout variant="large">
          <div>children</div>
        </Layout>
      )
      .toJSON();
    expect(layoutCmp).toMatchSnapshot();
  });

  it('should set small class with small variant', (): void => {
    const layoutCmp = renderer
      .create(
        <Layout variant="small">
          <div>children</div>
        </Layout>
      )
      .toJSON();
    expect(layoutCmp).toMatchSnapshot();
  });
});
