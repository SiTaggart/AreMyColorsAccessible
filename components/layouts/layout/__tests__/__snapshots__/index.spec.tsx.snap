// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
}

<div
    class="emotion-0 emotion-1"
  >
    <div>
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`Layout should set full class with full variant 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
  max-width: 95%;
}

<div
    class="emotion-0 emotion-1"
  >
    <div>
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`Layout should set large class with large variant 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
  max-width: 1200px;
}

<div
    class="emotion-0 emotion-1"
  >
    <div>
      children
    </div>
  </div>
</DocumentFragment>
`;

exports[`Layout should set small class with small variant 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
  max-width: 1150px;
  padding: 0 1rem;
}

<div
    class="emotion-0 emotion-1"
  >
    <div>
      children
    </div>
  </div>
</DocumentFragment>
`;
